### DISCUSSION TITLE
Definition of "$EXISTS" filter for REST, Documents API

### DISCUSSION THREAD

**Discussion by tatu-at-datastax** [stargate/stargate#1519] [RFCs]
It looks like operator "$EXISTS" has two meanings within Stargate 1.0 APIs:

* In Documents API it is similar to MongoDB's operator (see https://docs.mongodb.com/manual/reference/operator/query/exists/), that is, it allows for selecting documents that have a value for given property ("property X exists with a value")
* In REST API it appears to be an alias for "boolean property X has value 'true'" (that is, same as `$EQ`). This seems odd as such usage is superfluous so no separate operator would be needed

So which definition is correct one? Is it ok for 2 APIs to have different interpretation now, or in future (with Stargate V2 work we could revisit this)?

One significant issue here is that this operator/filter is only tested for Documents API; REST API behavior is only loosely indicated by Swagger comments for relevant REST endpoint (issue uncovered with #1518).

**Comment by olim7t** [comment]
IHMO the best solution is to deprecate it in the REST API:
- as you mentioned it's superfluous
- it has no equivalent in CQL, so not something that users coming from a Cassandra background will be looking for.
- it's super confusing: why just boolean? If true is the only value, why do you still have to specify it?
- if we wanted semantics closer to the Documents API, the best approximation would be whether the column is null. But that's not a valid query in CQL.

So I would either document that it's a synonym for `$eq`, or remove it from the Swagger docs altogether.

**Reply by tatu-at-datastax** [reply]
I agree with this, given that implementing behavior similar to Documents API is probably not technically feasible (if I understand things correctly).
